server/src/com/cloud/dc/dao/DedicatedResourceDaoImpl.java

```diff
@@ -312,6 +317,65 @@ public DedicatedResourceVO findByHostId(Long hostId) {
     }
 
     @Override
+    public List<DedicatedResourceVO> listAvailableResources(Long accountId, Long... domains) {
```

Comment:
@anshul1886 Check out SearchBuilder class, it has methods like or(), op(), cp() which can be used to construct the query. Also the same field can be used in multiple places, search for reference to or(), op() etc. elsewhere in the code and you will find how to use them. One such place is QueryManagerImpl.java